pymc3/tests/test_distributions_random.py

```diff
@@ -485,43 +399,391 @@ class TestMoyal(BaseTestCases.BaseTestCase):
+        assert len(self.expected_rv_op_params) == len(aesera_dist_inputs)
+        for (expected_name, expected_value), actual_variable in zip(
+            self.expected_rv_op_params.items(), aesera_dist_inputs
+        ):
+            assert_almost_equal(expected_value, actual_variable.eval(), decimal=self.decimal)
+
+    def check_rv_size(self):
+        # test sizes
+        sizes_to_check = self.sizes_to_check or [None, (), 1, (1,), 5, (4, 5), (2, 4, 2)]
+        sizes_expected = self.sizes_expected or [(), (), (1,), (1,), (5,), (4, 5), (2, 4, 2)]
+        for size, expected in zip(sizes_to_check, sizes_expected):
+            actual = change_rv_size(self.pymc_rv, size).eval().shape
+            assert actual == expected
+
+        # test negative sizes raise
+        for size in [-2, (3, -2)]:
+            with pytest.raises(ValueError):
+                change_rv_size(self.pymc_rv, size).eval()
+
+        # test multi-parameters sampling for univariate distributions
+        if self.pymc_dist.rv_op.ndim_supp == 0:
+            params = {
+                k: p * np.ones(self.repeated_params_shape) for k, p in self.pymc_dist_params.items()
+            }
+            self._instantiate_pymc_rv(params)
+            sizes_to_check = [None, self.repeated_params_shape, (5, self.repeated_params_shape)]
+            sizes_expected = [
+                (self.repeated_params_shape,),
+                (self.repeated_params_shape,),
+                (5, self.repeated_params_shape),
+            ]
+            for size, expected in zip(sizes_to_check, sizes_expected):
+                actual = change_rv_size(self.pymc_rv, size).eval().shape
+                assert actual == expected
+
+    def sanity_check_tests_to_run(self):
+        assert len(self.tests_to_run) == len(
+            set(self.tests_to_run)
+        ), "There are duplicates in the list of tests_to_run"
+
+
+def seeded_scipy_distribution_builder(dist_name: str) -> Callable:
+    return lambda self: functools.partial(
+        getattr(st, dist_name).rvs, random_state=self.get_random_state()
+    )
 
-@pytest.mark.skip(reason="This test is covered by Aesara")
-class TestDirichlet(SeededTest):
-    @pytest.mark.parametrize(
-        "shape, size",
-        [
-            ((2), (1)),
-            ((2), (2)),
-            ((2, 2), (2, 100)),
-            ((3, 4), (3, 4)),
-            ((3, 4), (3, 4, 100)),
-            ((3, 4), (100)),
-            ((3, 4), (1)),
-        ],
+
+def seeded_numpy_distribution_builder(dist_name: str) -> Callable:
+    return lambda self: functools.partial(
+        getattr(np.random.RandomState, dist_name), self.get_random_state()
+    )
+
+
+class TestDiscreteWeibull(BaseTestDistribution):
+    def discrete_weibul_rng_fn(self):
+        p = seeded_numpy_distribution_builder("uniform")
+        return (
+            lambda size, q, beta: np.ceil(
+                np.power(np.log(1 - p(self)(size=size)) / np.log(q), 1.0 / beta)
```

Comment:
I wounder whether this can be simplified / made more readable? the `p(self)(size=size)` looks very opaque

Reply:
yeah, it was not great. I did a minor change, it should be better now. Let me know if you have a better alternative in mind